firewall/handlers/whitelist_handler.go

```diff
@@ -0,0 +1,50 @@
+package handlers
+
+import (
+	"errors"
+)
+
+type WhitelistHandler struct {
+	whiteQueries[] string
+}
+
+var ErrQueryNotInWhitelist = errors.New("query not in whitelist")
+
+func NewWhitelistHandler(whiteQueries []string) (*WhitelistHandler, error) {
+
+	return &WhitelistHandler{whiteQueries:whiteQueries}, nil
+}
+
+func(handler * WhitelistHandler) CheckQuery(query string) error {
+	if !handler.contains(query) {
+		return ErrQueryNotInWhitelist
+	}
+	return nil
+}
+
+func(handler * WhitelistHandler) AddQueriesToWhitelist(queries []string) {
+
+	for index := 0; index < len(queries); index++ {
+		handler.whiteQueries = append(handler.whiteQueries, queries[index])
+	}
+	removeDuplicates(&handler.whiteQueries)
+}
+
+func (handler * WhitelistHandler) RemoveQueriesFromWhitelist(queries []string) {
```

Comment:
i think we should add test for this function

Reply:
It is implicitly tested within main firewall tests of black/white list